### PR TITLE
Recycling

### PR DESCRIPTION
**Pull request by linuskar**
Recycling machine

- Recycling generates money
- Counting the trash recycled
- Recycling is based on the trash that is left to recycle

### CODE REVIEW

**Comment by linuskar** [comment]
Ruleset not working for older sprint branches?

**Comment by linuskar** [comment]
> Ruleset not working for older sprint branches?

image

**Comment by Oskmey** [comment]
Yes, it should work now. The issue before was that I was using regular regex, but I needed to use fnmatch syntax

**Comment by Oskmey** [comment]
Yes, it should work now. The issue before was that I was using regular regex, but I needed to use fnmatch syntax

File: DAT257AgileProject/Assets/Scripts/GameManager.cs [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+    private int recycledTrashCount;
+
+    // Start is called before the first frame update
+    void Start()
+    {
+        playerStatsManager = GameObject.FindGameObjectWithTag("Player").GetComponent<PlayerStatsManager>();
+        recyclingManager = GameObject.FindGameObjectWithTag("Recycling Manager").GetComponent<RecyclingManager>();
+
+        // TEMP: adding trash to recycle
+        GenerateTrash(2);
```
Comment (Oskmey) [inline DAT257AgileProject/Assets/Scripts/GameManager.cs:29]: We might overlook removing this when putting everything together.

File: DAT257AgileProject/Assets/Scripts/Player/PlayerStatsManager.cs [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+    {
+        get { return money; }
+        set { money = value; }
+    }
+
+    // Start is called before the first frame update
+    void Awake()
+    {
+        recycledTrashList = new();
+        money = 0;
```
Comment (Oskmey) [inline DAT257AgileProject/Assets/Scripts/Player/PlayerStatsManager.cs:31]: I'm not certain, but will this be implemented in the player class later? Could then we add something that stores it between scenes. :zany_face:

Comment (Oskmey) [reply DAT257AgileProject/Assets/Scripts/Player/PlayerStatsManager.cs:31]: This seems like something we could add when merging everything

**Review by Oskmey — CHANGES_REQUESTED** [review]
You can just resolve my comments if you think they are not good.

#  Looks very good!

**Review by Caramajau — CHANGES_REQUESTED** [review]
### Looks good in general, however...

### Some minor things that should be changed are:

Spelling mistakes:
- GameManager at line 62 (should be recyclable)
- GameManager at line 69 (should be success)

Better encapsulation in RecyclingManager
- From what I can see the properties in RecyclingManager do not need to have setters, their values are only changed locally in the class or not at all. 
- Also IReadOnlyList as a return type can be used for better encapsulation. 

Other not important spelling mistakes (can ignore if you want):
- RecyclingTests at line 24 (should be recyclable)
- RecyclingTests at line 37 (should be recyclable)
- Recycling machine at line 39 (should be recyclable)

File: DAT257AgileProject/Assets/Scripts/GameManager.cs (hunk 1 above)
Comment (linuskar) [reply DAT257AgileProject/Assets/Scripts/GameManager.cs:29]: yes

Comment (linuskar) [reply DAT257AgileProject/Assets/Scripts/GameManager.cs:29]: > We might overlook removing this when putting everything together.

Yes

**Comment by linuskar** [comment]
> ### Looks good in general, however...
> ### Some minor things that should be changed are:
> Spelling mistakes:
> 
> * GameManager at line 62 (should be recyclable)
> * GameManager at line 69 (should be success)
> 
> Better encapsulation in RecyclingManager
> 
> * From what I can see the properties in RecyclingManager do not need to have setters, their values are only changed locally in the class or not at all.
> * Also IReadOnlyList as a return type can be used for better encapsulation.
> 
> Other not important spelling mistakes (can ignore if you want):
> 
> * RecyclingTests at line 24 (should be recyclable)
> * RecyclingTests at line 37 (should be recyclable)
> * Recycling machine at line 39 (should be recyclable)

Commited based on changes

**Review by Caramajau — APPROVED** [review]

**Review by Oskmey — APPROVED** [review]

**Pull request merged**